tests/suites/test_suite_psa_crypto.function

```diff
@@ -6747,6 +7002,144 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE depends_on:MBEDTLS_ECP_RESTARTABLE */
+void sign_verify_hash_interruptible( int key_type_arg, data_t *key_data,
```

Comment:
[non-blocker] I think we should briefly explain the purpose of this function, especially what it brings that the others don't. (Actually that's probably not specific to this function, and something we've historically not done enough, but we should make a conscious effort to improve.)

I assume that the point is that `sign_hash_interruptible` only works for deterministic ECDSA as it compares the generated signature to a known answer (which btw should be documented there too), so we need something to validate that we generate valid signatures with non-deterministic ECDSA, and this is what this function does, by using our verify function. (This is probably not obvious to someone who looks at tests of non-deterministic primitives for the first time, especially as we have a mix of deterministic and non-deterministic here.)

Edit: also, being explicit about the goals helps keeping the function focused. If this is indeed about testing non-deterministic signature generation as I assume, then we'd want to be very thorough about the signing part, that is, we should add test for `num_ops` as this has not been tested elsewhere yet. OTOH, if we're using verify here only to check that the generated signature is correct, we don't need to check its `num_complete`, as that's already done in `verify_hash_interruptible` - we could even use non-interruptible `verify()` here is we just want to validate that the generated signature is correct. (Also, IMO the part where we corrupt the signature and check that verify fails belongs in `verify_hash_interruptible` rather than here - again, if my assumption about the purpose of this function is correct.)

Reply:
#7115